compendium/DeclarativeServices/test/bench/CMakeLists.txt

```diff
@@ -0,0 +1,214 @@
+set(_declarativeservices_benchmark_tests
+  GetDSServiceTest.cpp
+)
+
+include_directories(${PROJECT_BINARY_DIR}/include
+  ${CppMicroServices_SOURCE_DIR}/compendium/CM/include
+  ${CppMicroServices_BINARY_DIR}/compendium/CM/include
+  )
+
+set(_additional_srcs
+  ../TestUtils.cpp
+  ../../src/SCRActivator.cpp
+  )
+
+#-----------------------------------------------------------------------------
+# Build the main test driver executable
+#-----------------------------------------------------------------------------
+
+# Generate a custom "bundle init" file for the test driver executable
+usFunctionGenerateBundleInit(
+  TARGET ${us_declarativeservices_bench_test_exe_name}
+  OUT _additional_srcs
+  )
+
+usFunctionGetResourceSource(
+  TARGET ${us_declarativeservices_bench_test_exe_name}
+  OUT _additional_srcs)
+
+add_executable(${us_declarativeservices_bench_test_exe_name}
+  ${_declarativeservices_benchmark_tests}
+  ${_additional_srcs})
+
+if (US_COMPILER_MSVC AND BUILD_SHARED_LIBS)
+  target_compile_options(${us_declarativeservices_bench_test_exe_name} PRIVATE -DGTEST_LINKED_AS_SHARED_LIBRARY)
+endif()
+
+set_property(
+  TARGET ${us_declarativeservices_bench_test_exe_name}
+  APPEND PROPERTY COMPILE_DEFINITIONS US_BUNDLE_NAME=main
+  )
+
+set_property(
+  TARGET ${us_declarativeservices_bench_test_exe_name}
+  PROPERTY US_BUNDLE_NAME main
+  )
+
+target_include_directories(${us_declarativeservices_bench_test_exe_name}
+  PRIVATE $<TARGET_PROPERTY:util,INCLUDE_DIRECTORIES>)
+
+# Disable deprecation warnings.
+if (("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang") OR
+    ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang") OR
+    ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU"))
+  set_source_files_properties(${_declarativeservices_benchmark_tests}
+    PROPERTIES
+    COMPILE_FLAGS -Wno-error=deprecated-declarations
+    COMPILE_FLAGS -Wno-sign-compare
+    )
+endif()
+
+set(_test_bundles
+  BenchmarkDS
+  DSFrenchDictionary
+  DSGraph01
+  DSGraph02
+  DSGraph03
+  DSGraph04
+  DSGraph05
+  DSGraph06
+  DSGraph07
+  DSSpellChecker
+  TestBindUnbindThrows
+  TestBundleDSa
+  TestBundleDSb
+  TestBundleDSDGMU
+  TestBundleDSDGOU
+  TestBundleDSDRMU
+  TestBundleDSDROU
+  TestBundleDSSLE1
+  TestBundleDSSLE2
+  TestBundleDSTOI1
+  TestBundleDSTOI2
+  TestBundleDSTOI3
+  TestBundleDSTOI5
+  TestBundleDSTOI6
+  TestBundleDSTOI7
+  TestBundleDSTOI8
+  TestBundleDSTOI9
+  TestBundleDSTOI10
+  TestBundleDSTOI12
+  TestBundleDSTOI14
+  TestBundleDSTOI15
+  TestBundleDSTOI16
+  TestBundleDSTOI18
+  TestBundleDSTOI19
+  TestBundleDSCA01
+  TestBundleDSCA02
+  TestBundleDSCA03
+  TestBundleDSCA04
+  TestBundleDSCA05
+  TestBundleDSCA05a
+  TestBundleDSCA07
+  TestBundleDSCA08
+  TestBundleDSCA09
+  TestBundleDSCA12
+  TestBundleDSCA16
+  TestBundleDSCA20
+  TestBundleDSCA24
+  TestBundleDSCA26
+  TestBundleDSCA27
+  EnglishDictionary
+  )
+
+target_link_libraries(${us_declarativeservices_bench_test_exe_name}
```

Comment:
This should only contain the link time dependencies needed currently. e.g. I don't think the dictionary and spell checker services are required.